content/en/docs/v3.5/op-guide/data_corruption.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Data Corruption
+weight: 5000
+description: etcd data corruption and recovery
+---
+
+etcd has built in automated data corruption detection to prevent member state from diverging.
+
+## Enabling data corruption detection
+
+Data corruption detection can be done in two ways:
+* Initial check, enabled with `--experimental-initial-corrupt-check` flag.
+* Periodic check, enabled with `--experimental-corrupt-check-time` flag.
+
+Initial check will be executed during bootstrap of etcd member.
+Member will compare it's persistent state vs other members and exit if there is a mismatch.
+
+Periodic check will be executed by the cluster leader in a cluster that is already running.
+Leader will compare it's persistent state vs other members and raise a CORRUPT ALARM if there is a mismatch.
+Period of checks is configured using format: `1m` - every minute, `1h` - evey hour.
+Recommended period is a couple of hours as there is a high performance cost.
+
+## Restoring a corrupted member
+
+There are three ways to restore a corrupted member:
+* Purge member persistent state
+* Replace member
+* Restore whole cluster
+
+After the corrupted member is restored, CORRUPT ALARM can be removed.
+
+### Purge member persistent state
+
+Members state can be purged by removing the `snap` directory within etcd data directory.
```

Comment:
can by purged by: 
  - stopping the etcd instance
  - backing up entire etcd data directory
  - moving out the `snap` subdirectory from the etcd `datadir`
  - starting `etcd` again. It's expected it will download up-to-data snapshot from the leader.
   
  (Marek: I assume during restart of etcd you need to pass all the flags pointing on peers and cluster_id), right ? If so - we should write it explicitly.

Reply:
I might remember this wrong, but I doubt WAL contains information about memberId->URL mapping. 
It contains incremental information about changes... but it's not something that should be considered trust-worthy or periodically snapshot.... I thing WAL snapshot contains only list of memberIds without URLs.